File: Professional/ActiveReportsWithMVC/C#/ActiveReportsWithMVC/DBContext/NWindData.cs
using System;
using System.Collections.Generic;
using System.Data.OleDb;
using GrapeCity.ActiveReports.Samples.ActiveReportsWithMVC.Models;
using System.Configuration;
namespace GrapeCity.ActiveReports.Samples.ActiveReportsWithMVC.DBContext
{
	/// <summary>
	/// Gets the entire Customer,Orders,Order Details table of the NWIND.mdb
	/// Populates them into in-memory objects
	/// Analogous to how you would fetch data via Ado.net entity framework 
	/// </summary>
	public static class NWindData
	{
		public static List<Customer> Customers;
		public static List<Order> Orders;
		public static List<OrderDetails> OrderDetails;
		
		/// <summary>
		/// Gets customers from Customers table of NWind.mdb
		/// </summary>
		/// <returns>list of Customers</returns>
		public static List<Customer> GetNWindCustomers()
		{
			if (Customers == null)
			{
				var allcustomers = new List<Customer>();
				var conn = new OleDbConnection(GetNWindConnectString());
				conn.Open();
				var command = new OleDbCommand("Select Top 20 * from Customers", conn);
				var dr = command.ExecuteReader();
				while (dr != null && dr.Read())
				{
					allcustomers.Add(new Customer
					{
						Address = dr["Address"].ToString(),
						City = dr["City"].ToString(),
						CompanyName = dr["CompanyName"].ToString(),
						ContactName = dr["ContactName"].ToString(),
						ContactTitle = dr["ContactTitle"].ToString(),
						Country = dr["Country"].ToString(),
						CustomerId = dr["CustomerID"].ToString(),
						Fax = dr["Fax"].ToString(),
						Phone = dr["Phone"].ToString(),
						PostalCode = dr["PostalCode"].ToString(),
						Region = dr["Region"].ToString()
					});
				}
				conn.Close();
				Customers = allcustomers;
			}
			
			return Customers;
		}
		/// <summary>
		/// Gets Orders from Nwind.mdb
		/// </summary>
		/// <returns>list of orders</returns>
		public static List<Order> GetNWindOrders()
		{
			if (Orders == null)
			{
				var allorders = new List<Order>();
				var conn = new OleDbConnection(GetNWindConnectString());
				conn.Open();
				var command = new OleDbCommand("Select * from Orders", conn);
				var dr = command.ExecuteReader();
				while (dr != null && dr.Read())
				{
					allorders.Add(new Order
					  {
						  CustomerId = dr["CustomerID"].ToString(),
						  EmployeeId = Convert.ToInt64(dr["EmployeeID"]),
						  Freight = Convert.ToDecimal(dr["Freight"]),
						  OrderDate = dr["OrderDate"] == DBNull.Value ? DateTime.Now.ToShortDateString() : Convert.ToDateTime(dr["OrderDate"]).ToShortDateString(),
						  OrderId = Convert.ToInt64(dr["OrderID"]),
						  RequiredDate = dr["RequiredDate"] == DBNull.Value ? DateTime.Now.ToShortDateString() : Convert.ToDateTime(dr["RequiredDate"]).ToShortDateString(),
						  ShipAddress = dr["ShipAddress"].ToString(),
						  ShipCity = dr["ShipCity"].ToString(),
						  ShipCountry = dr["ShipCountry"].ToString(),
						  ShipName = dr["ShipName"].ToString(),
						  ShippedDate = dr["ShippedDate"] == DBNull.Value ? DateTime.Now.ToShortDateString() : Convert.ToDateTime(dr["ShippedDate"]).ToShortDateString(),
						  ShipPostalCode = dr["ShipPostalCode"].ToString(),
						  ShipRegion = dr["ShipRegion"].ToString(),
						  ShipVia = Convert.ToInt64(dr["ShipVia"])
					  });
				}
				conn.Close();
				Orders = allorders;
			}
			return Orders;
			 
		}
		/// <summary>
		/// Gets OrdersDetails from Order Details table of NWind.mdb
		/// </summary>
		/// <returns>List of OrderDetails</returns>
		public static List<OrderDetails> GetOrderDetails()
		{
			
			if (OrderDetails == null)
			{
				var allorderdetails = new List<OrderDetails>();
				var conn = new OleDbConnection(GetNWindConnectString());
				conn.Open();
				var command = new OleDbCommand("Select * from [Order Details]", conn);
				var dr = command.ExecuteReader();
				while (dr != null && dr.Read())
				{
					allorderdetails.Add(new OrderDetails { 
					Discount=Convert.ToInt32(dr["Discount"]),
					OrderId = Convert.ToInt64(dr["OrderID"]),
					ProductId = Convert.ToInt32(dr["ProductID"]),
					Quantity = Convert.ToInt32(dr["Quantity"]),
					UnitPrice = Convert.ToDecimal(dr["UnitPrice"])
					});
				}
				conn.Close();
				OrderDetails = allorderdetails;
			}
			return OrderDetails;
		}
		/// <summary>
		/// Reads the ConnectionString for the NWind.mdb from the web.config file
		/// </summary>
		/// <returns>connectionstring:-"NWindConnectionString" from web.config</returns>
		private static string GetNWindConnectString()
		{
			return ConfigurationManager.ConnectionStrings["NWindConnectionString"].ConnectionString;
		}
	}
}

File: Professional/ActiveReportsWithMVC/VB.NET/ActiveReportsWithMVC/DBContext/NWindData.vb
Imports System
Imports System.Collections.Generic
Imports System.Data.OleDb
Imports GrapeCity.ActiveReports.Samples.ActiveReportsWithMVC.Models
Namespace DBContext
	'Gets the entire Customer,Orders,Order Details table of the NWIND.mdb
	'Populates them into in-memory objects
	'Analogous to how you would fetch data via Ado.net entity framework 
	Public NotInheritable Class NWindData
		Private Sub New()
		End Sub
		Public Shared Customers As List(Of Customer)
		Public Shared Orders As List(Of Order)
		Public Shared OrderDetails As List(Of OrderDetails)
		'Gets customers from Customers table of NWind.mdb
		Public Shared Function GetNWindCustomers() As List(Of Customer)
			If Customers Is Nothing Then
				Dim allcustomers As New List(Of Customer)()
				Dim conn As New OleDbConnection("Provider=Microsoft.Jet.OLEDB.4.0;Data Source=C:\Users\evan.warren\Documents\GrapeCity Samples\ActiveReports 12\Data\NWIND.mdb;Persist Security Info=False")
				conn.Open()
				Dim command As New OleDbCommand("Select Top 20 * from Customers", conn)
				Dim dr As OleDbDataReader = command.ExecuteReader()
				While dr.Read()
					allcustomers.Add(New Customer() With { _
					 .Address = dr("Address").ToString(), _
					 .City = dr("City").ToString(), _
					 .CompanyName = dr("CompanyName").ToString(), _
					 .ContactName = dr("ContactName").ToString(), _
					 .ContactTitle = dr("ContactTitle").ToString(), _
					 .Country = dr("Country").ToString(), _
					 .CustomerId = dr("CustomerID").ToString(), _
					 .Fax = dr("Fax").ToString(), _
					 .Phone = dr("Phone").ToString(), _
					 .PostalCode = dr("PostalCode").ToString(), _
					 .Region = dr("Region").ToString() _
					})
				End While
				conn.Close()
				Customers = allcustomers
			End If
			Return Customers
		End Function
		'Gets Orders from Nwind.mdb
		Public Shared Function GetNWindOrders() As List(Of Order)
			If Orders Is Nothing Then
				Dim allorders As New List(Of Order)()
				Dim conn As New OleDbConnection("Provider=Microsoft.Jet.OLEDB.4.0;Data Source=C:\Users\evan.warren\Documents\GrapeCity Samples\ActiveReports 12\Data\NWIND.mdb;Persist Security Info=False")
				conn.Open()
				Dim command As New OleDbCommand("Select * from Orders", conn)
				Dim dr As OleDbDataReader = command.ExecuteReader()
				While dr.Read()
					allorders.Add(New Order() With { _
					 .CustomerId = dr("CustomerID").ToString(), _
					 .EmployeeId = Convert.ToInt64(dr("EmployeeID")), _
					 .Freight = Convert.ToDecimal(dr("Freight")), _
					 .OrderDate = If(dr("OrderDate") Is DBNull.Value, DateTime.Now.ToShortDateString(), Convert.ToDateTime(dr("OrderDate")).ToShortDateString()), _
					 .OrderId = Convert.ToInt64(dr("OrderID")), _
					 .RequiredDate = If(dr("RequiredDate") Is DBNull.Value, DateTime.Now.ToShortDateString(), Convert.ToDateTime(dr("RequiredDate")).ToShortDateString()), _
					 .ShipAddress = dr("ShipAddress").ToString(), _
					 .ShipCity = dr("ShipCity").ToString(), _
					 .ShipCountry = dr("ShipCountry").ToString(), _
					 .ShipName = dr("ShipName").ToString(), _
					 .ShippedDate = If(dr("ShippedDate") Is DBNull.Value, DateTime.Now.ToShortDateString(), Convert.ToDateTime(dr("ShippedDate")).ToShortDateString()), _
					 .ShipPostalCode = dr("ShipPostalCode").ToString(), _
					 .ShipRegion = dr("ShipRegion").ToString(), _
					 .ShipVia = Convert.ToInt64(dr("ShipVia")) _
					})
				End While
				conn.Close()
				Orders = allorders
			End If
			Return Orders
		End Function
		'Gets OrdersDetails from Order Details table of NWind.mdb
		Public Shared Function GetOrderDetails() As List(Of OrderDetails)
			If OrderDetails Is Nothing Then
				Dim allorderdetails As New List(Of OrderDetails)()
				Dim conn As New OleDbConnection("Provider=Microsoft.Jet.OLEDB.4.0;Data Source=C:\Users\evan.warren\Documents\GrapeCity Samples\ActiveReports 12\Data\NWIND.mdb;Persist Security Info=False")
				conn.Open()
				Dim command As New OleDbCommand("Select * from [Order Details]", conn)
				Dim dr As OleDbDataReader = command.ExecuteReader()
				While dr.Read()
					allorderdetails.Add(New OrderDetails() With { _
					 .Discount = Convert.ToInt32(dr("Discount")), _
					 .OrderId = Convert.ToInt64(dr("OrderID")), _
					 .ProductId = Convert.ToInt32(dr("ProductID")), _
					 .Quantity = Convert.ToInt32(dr("Quantity")), _
					 .UnitPrice = Convert.ToDecimal(dr("UnitPrice")) _
					})
				End While
				conn.Close()
				OrderDetails = allorderdetails
			End If
			Return OrderDetails
		End Function
	End Class
End Namespace
